components/List/src/List.tsx

```diff
@@ -1,33 +1,28 @@
-import React from 'react';
-import { List as MaterialList } from '@material-ui/core';
-import BaseProps from '@gemeente-denhaag/baseprops';
+import React, { HTMLAttributes } from 'react';
 import { ListSubheaderProps } from './ListSubheader';
-import { list_classes as classes } from './bem-mapping';
+import clsx from 'clsx';
 
-import './mui-override.scss';
-import './list-item.scss';
-import './list-item-secondary-action.scss';
-import './list-item-icon.scss';
-import './list-item-text.scss';
-import './list-subheader.scss';
+import './list.scss';
 
-export interface ListProps extends Omit<BaseProps, 'classes'> {
+export interface ListProps extends HTMLAttributes<HTMLDivElement> {
   /**
    * The content of the subheader, normally `ListSubheader`.
    */
-  subheader: React.ReactElement<ListSubheaderProps>;
+  subheader?: React.ReactElement<ListSubheaderProps>;
 }
 
 /**
  * Primary UI component for user interaction
  */
-export const List: React.FC<ListProps> = (props: ListProps) => {
-  const materialProps = { subheader: props.subheader };
-
+export const List: React.FC<ListProps> = ({ subheader, ...props }: ListProps) => {
+  const rootClassNames = clsx('denhaag-list__wrapper', {
+    'denhaag-list__wrapper--with-subheader': subheader,
+  });
   return (
-    <MaterialList {...materialProps} classes={classes}>
-      {props.children}
-    </MaterialList>
+    <div {...props} className={rootClassNames}>
+      {subheader}
+      <ul className="denhaag-list">{props.children}</ul>
+    </div>
```

Comment:
I'm not sure how I feel about this BEM structure. There is an element selector (`denhaag-list__wrapper`) higher than the component block (`denhaag-list`). This makes it feel as thought the List header is not part of the component.